File: gmao_backend/src/main/java/com/gmao/backend/entity/Equipe.java
package com.gmao.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "equipes")
@com.fasterxml.jackson.annotation.JsonIgnoreProperties({ "hibernateLazyInitializer", "handler" })
public class Equipe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String nom;

    private String description;

    @OneToMany(mappedBy = "equipe")
    @com.fasterxml.jackson.annotation.JsonIgnoreProperties({ "equipe", "password", "role", "authorities",
            "notificationPreferences", "hibernateLazyInitializer", "handler" })
    private List<User> membres;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chef_id")
    @com.fasterxml.jackson.annotation.JsonIgnoreProperties({ "equipe", "password", "role", "authorities",
            "notificationPreferences", "hibernateLazyInitializer", "handler" })
    private User chef;
}

File: gmao_mobile/lib/models/equipe.dart
import 'user.dart';

class Equipe {
  final int id;
  final String nom;
  final String? description;
  final User? chef;
  final List<User>? membres;

  Equipe({
    required this.id,
    required this.nom,
    this.description,
    this.chef,
    this.membres,
  });

  factory Equipe.fromJson(Map<String, dynamic> json) {
    return Equipe(
      id: json['id'],
      nom: json['nom'],
      description: json['description'],
      chef: json['chef'] != null ? User.fromJson(json['chef']) : null,
      membres: json['membres'] != null
          ? (json['membres'] as List).map((e) => User.fromJson(e)).toList()
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nom': nom,
      'description': description,
      'chef': chef != null ? {'id': chef!.id} : null,
      'membres': membres?.map((e) => e.toJson()).toList(),
    };
  }
}
